lib/puppet/provider/network_config/interfaces.rb

```diff
@@ -187,7 +187,18 @@ def self.parse_file(_filename, contents)
           # Instance[name].name   = name
           Instance[name].family = family
           Instance[name].method = method
-          Instance[name].mode   = :raw
+          # for the vlan naming conventions (a mess), see
+          # man 5 vlan-interfaces
+          case name
+            # 'vlan22'
+            when /^vlan/
+              Instance[name].mode   = :vlan
+            # 'eth2.0003' or 'br1.2'
+            when /^[a-z]{1,}[0-9]{1,}\.[0-9]{1,4}/
```

Comment:
@bastelfreak this regex is quite straightforward and matches with the string in the comment above.